content/contributing.md

```diff
@@ -0,0 +1,126 @@
+
+Our notebooks do not support getting input from the reader yet. Although it is an idea we consider for future, please keep narrative notebooks for now.
+
++++ {"id": "6TqY5A99JgXc"}
+
+### 5. Do not forget to add path of your notebook to `index.md`.
+
+You should include the path of your notebook in index.md file under `nx-guides/content/algorithms`.
+
++++ {"id": "o8wvepdRKPFK"}
+
+### 6. Header Levels
+
+Header levels should be incremented one by one. If you jump from level 2 to level 4 header, for example, msyt will produce an error to prevents you from passing the tests. In this example, if the current header level is 2, the following header level needs to be either 2 or 3.
+
++++ {"id": "sHw3aPpOKfJU"}
+
+### 7. You do not need to implement the algorithm in the same exact way as done inside NetworkX.
+
+nx-guides provides a pedagogical source for NetworkX algorithms. For this, you do not have to include source code of the algorithm as it is under NetworkX. If possible, feel free to remove bits that you think can be better compressed :)
+
++++ {"id": "78u0INXfL47X"}
+
+### 8. Feel free to use real-world datasets
+
+One of the aims of nx-guides notebooks is to use different algorithms to explore and analyse real world datasets. Feel free to use them if you believe it is useful.
+
+Here is a good source for datasets:
+
+http://snap.stanford.edu/data/index.html
+
++++
+
+### 9. What if the tests are still failing?
+
+Once all tests are completed, you can see warnings and errors that prevents your PR from passing the tests. To do that, go to the bottom of "Conversation" page in your PR. There will be red cross signs on the left side of "ci/circleci: build-docs" test suite. Click on the "Details" link on the right side of it to see errors and warnings.
+
+You can also click on the "Details" link on the right side of "ci/circleci: build-docs artifact". If your notebook is built, this will bring you to the full documentation for the project as if this branch was merged. You can then navigate to the notebook you have created and check that your documentation looks good.
+
+### Environment
+
+A good way to go about editing your markdown file is with Jupyter Notebook or other markdown file editors. Just make sure the metadata fits that of the other markdown files in nx-guides. 
+
++++
+
+## Format Guidelines
+
+### 1. Write a clear title.
+### 2. When introducing your topic, reference what problem you are solving and what part of NetworkX you are using.
+### 3. Include an "import packages" section at the top of the notebook after your introduction.
+
+Add all import statements for packages in a code cell underneath your introduction (see other notebooks for examples)
+### 4. Comment code heavily, especially confusing sections. 
+
+### 5. Cite all sources. Include in-text references and a references section at the bottom of the document.
+
+For examples of how to write these references and cite, see other notebooks
```

Comment:
The font size is too large for the list of "Formatting Guidelines" near the end. 
I think it might be better to use the list style as is done for the roadmap numbered items at the top of the page.
```suggestion
1. Write a clear title.
2. When introducing your topic, reference what problem you are solving and what part of NetworkX you are using.
3. Include an "import packages" section at the top of the notebook after your introduction.
Add all import statements for packages in a code cell underneath your introduction (see other notebooks for examples)
4. Comment code heavily, especially confusing sections. 
5. Cite all sources. Include in-text references and a references section at the bottom of the document.
For examples of how to write these references and cite, see other notebooks
```